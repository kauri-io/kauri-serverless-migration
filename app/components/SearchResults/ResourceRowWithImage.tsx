import moment from 'moment'
import styled from 'styled-components'
import UserAvatar from '../UserAvatar'
import { Label, H1, BodyCard } from '../Typography'
import TagList from '../Tags/TagList'
import Image from '../Image'
import theme from '../../lib/theme-config'
import slugify from 'slugify'

const DEFAULT_CARD_WIDTH = theme.DEFAULT_CARD_WIDTH

const ResourceRow = styled.div`
    display: flex;
    width: ${props => DEFAULT_CARD_WIDTH * 3 + props.theme.space[3] * 2}px;
    height: 195px;
    background-color: ${props => props.theme.colors.white};
    border-radius: 4px;
    box-shadow: 0 0 4px 0 rgba(0, 0, 0, 0.11);
`

const Content = styled.div`
    display: flex;
    flex-direction: column;
    > *:not(:last-child) {
        margin-bottom: ${props => props.theme.space[1]}px;
    }
    > *:nth-child(3) {
        margin-bottom: 0px;
    }
`

const Footer = Content

const Container = styled.section`
    display: flex;
    width: 100%;
    flex-direction: column;
    padding: ${props => props.theme.space[1]}px
        ${props => props.theme.space[2]}px;
`

const Divider = styled.div`
    width: 100%;
    margin-top: auto;
    margin-bottom: ${props => props.theme.space[1]}px;
    background-color: ${props => props.theme.colors.divider};
    height: 2px;
`

export interface IProps {
    id: string
    version?: number
    title: string
    description: string | null
    date: string
    imageURL: string | null
    tags: string[] | null
    username: string | null
    userId: string
    userAvatar: string | null
    linkComponent: (
        childrenProps: React.ReactElement<any>,
        route: string
    ) => React.ReactElement<any>
    ownerType: string // "USER" | "COMMUNITY" | "COLLECTION";
    resourceType: string // "article" | "community" | "collection"
}

const ResourceRowWithImage: React.SFC<IProps> = ({
    title,
    description,
    imageURL,
    id,
    resourceType,
    linkComponent,
    date,
    ownerType,
    tags,
    username,
    userId,
    userAvatar,
}) => {
    const slug = slugify(title, { lower: true })
    return (
        <ResourceRow key={id}>
            {imageURL &&
                linkComponent(
                    <Image
                        width={DEFAULT_CARD_WIDTH}
                        height={195}
                        image={imageURL}
                        borderTopLeftRadius={'4px'}
                        borderBottomLeftRadius={'4px'}
                    />,
                    resourceType === 'collection'
                        ? `/${slugify(title, { lower: true })}/${id}/c`
                        : resourceType === 'community'
                        ? `/community/${id}`
                        : `/${slug}/${id}/a`
                )}
            <Container>
                {linkComponent(
                    <Content>
                        <Label>
                            {ownerType === 'COLLECTION'
                                ? 'Updated ' +
                                  moment(date).format('DD MMM YYYY HH:mm')
                                : 'Posted ' +
                                  moment(date).format('DD MMM YYYY HH:mm')}
                        </Label>
                        <H1>{title && String(title)}</H1>
                        {description && (
                            <BodyCard>
                                {description && String(description)}
                            </BodyCard>
                        )}
                        {Array.isArray(tags) && tags.length > 0 && (
                            <TagList
                                maxTags={3}
                                color="textPrimary"
                                tags={tags}
                            />
                        )}
                    </Content>,
                    resourceType === 'collection'
                        ? `/${slugify(title, { lower: true })}/${id}/c`
                        : resourceType === 'community'
                        ? `/community/${id}`
                        : `/${slug}/${id}/a`
                )}
                <Divider />
                <Footer>
                    {linkComponent(
                        <UserAvatar
                            imageURL={imageURL}
                            cardType="ARTICLE"
                            fullWidth={true}
                            username={username}
                            userId={userId}
                            avatar={userAvatar}
                        />,
                        ownerType === 'COMMUNITY'
                            ? `/community/${userId}`
                            : `/public-profile/${userId}`
                    )}
                </Footer>
            </Container>
        </ResourceRow>
    )
}

export default ResourceRowWithImage
