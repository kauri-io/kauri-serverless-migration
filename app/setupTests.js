import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import theme from './lib/theme-config'
import { ThemeProvider } from 'styled-components'

Enzyme.configure({ adapter: new Adapter() })

const observeMock = {
    observe: () => null,
    disconnect: () => null, // maybe not needed
}

window.IntersectionObserver = () => observeMock

export const mountWithTheme = children =>
    mount(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const renderWithTheme = children =>
    render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const shallowWithTheme = children =>
    shallow(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
