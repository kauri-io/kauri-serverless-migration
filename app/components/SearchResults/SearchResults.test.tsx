import ResourceRowWithImage from './ResourceRowWithImage'
import { mountWithRedux } from '../../setupTests'

describe('components/ResourceRowWithImage', () => {
    it('should match snapshot', () => {
        const ResourceRowWithImageProps = {
            id: '123',
            title: 'test',
            description: 'test description',
            date: '2019-01-01',
            imageURL: 'test url',
            tags: ['test'],
            username: 'jimi',
            userId: '0x',
            userAvatar: 'test avatar',
            linkComponent: jest.fn(),
            ownerType: 'COMMUNITY',
            resourceType: 'ARTICLE',
        }

        const wrapper = mountWithRedux(
            <ResourceRowWithImage {...ResourceRowWithImageProps} />
        )
        expect(wrapper).toMatchSnapshot()
    })
})
