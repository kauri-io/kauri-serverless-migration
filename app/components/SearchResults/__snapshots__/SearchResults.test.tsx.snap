// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ResourceRowWithImage should match snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ThemeProvider
    theme={
      Object {
        "DEFAULT_CARD_WIDTH": 288,
        "MAIN_BODY_WIDTH": 1242,
        "bg": Object {
          "bgPrimary": "#1E2428",
          "bgSecondary": "#1E3D3B",
          "borderTextArea": "#979797",
          "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
          "disabledBackgroundColor": "#D6D6D6",
          "disabledTextColor": "#9B9B9B",
          "divider": "#EBEBEB",
          "errorRedColor": "#C03030",
          "hoverTextColor": "#108E72",
          "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
          "paddingTop": "2.5em",
          "primary": "#0BA986",
          "primaryColor": "#0BA986",
          "primaryDark": "#11856B",
          "primaryTextColor": "#1E2428",
          "secondary": "#11856B",
          "secondaryBlue": "rgb(81, 123, 162, 1)",
          "secondaryBlueDark": "rgb(81, 118, 142, 1)",
          "secondaryColor": "#1E3D3B",
          "secondaryTextColor": "#283035",
          "tertiaryBackgroundColor": "#F7F7F7",
          "textPrimary": "#1E2428",
          "white": "#FFFFFF",
        },
        "bgPrimary": "#1E2428",
        "bgSecondary": "#1E3D3B",
        "borderTextArea": "#979797",
        "breakpoints": Array [
          "500px",
          "52em",
          "1242px",
        ],
        "colors": Object {
          "bgPrimary": "#1E2428",
          "bgSecondary": "#1E3D3B",
          "borderTextArea": "#979797",
          "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
          "disabledBackgroundColor": "#D6D6D6",
          "disabledTextColor": "#9B9B9B",
          "divider": "#EBEBEB",
          "errorRedColor": "#C03030",
          "hoverTextColor": "#108E72",
          "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
          "paddingTop": "2.5em",
          "primary": "#0BA986",
          "primaryColor": "#0BA986",
          "primaryDark": "#11856B",
          "primaryTextColor": "#1E2428",
          "secondary": "#11856B",
          "secondaryBlue": "rgb(81, 123, 162, 1)",
          "secondaryBlueDark": "rgb(81, 118, 142, 1)",
          "secondaryColor": "#1E3D3B",
          "secondaryTextColor": "#283035",
          "tertiaryBackgroundColor": "#F7F7F7",
          "textPrimary": "#1E2428",
          "white": "#FFFFFF",
        },
        "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
        "disabledBackgroundColor": "#D6D6D6",
        "disabledTextColor": "#9B9B9B",
        "divider": "#EBEBEB",
        "errorRedColor": "#C03030",
        "fontSizes": Array [
          11,
          13,
          14,
          16,
          18,
          20,
          22,
          28,
          32,
        ],
        "fontWeight": Array [
          300,
          "normal",
          500,
          700,
        ],
        "hoverTextColor": "#108E72",
        "padContent": "
    max-width: 1242px;
    margin-left: auto;
    margin-right: auto;
    @media (min-width: 1242px) {
      min-width: 1242px;
    }
    @media (max-width: 1242px) {
      min-width: 90%;
    }
  ",
        "padding": "calc((100vw - 1242px) / 2)",
        "paddingTop": "2.5em",
        "primary": "#0BA986",
        "primaryColor": "#0BA986",
        "primaryDark": "#11856B",
        "primaryTextColor": "#1E2428",
        "secondary": "#11856B",
        "secondaryBlue": "rgb(81, 123, 162, 1)",
        "secondaryBlueDark": "rgb(81, 118, 142, 1)",
        "secondaryColor": "#1E3D3B",
        "secondaryTextColor": "#283035",
        "space": Array [
          0,
          10,
          15,
          30,
          60,
        ],
        "tertiaryBackgroundColor": "#F7F7F7",
        "textPrimary": "#1E2428",
        "white": "#FFFFFF",
      }
    }
  >
    <ResourceRowWithImage
      date="2019-01-01"
      description="test description"
      id="123"
      imageURL="test url"
      linkComponent={
        [MockFunction] {
          "calls": Array [
            Array [
              <Image
                borderBottomLeftRadius="4px"
                borderTopLeftRadius="4px"
                height={195}
                image="test url"
                width={288}
              />,
              "/test/123/a",
            ],
            Array [
              <ForwardRef(ResourceRowWithImage__Content)>
                <ForwardRef(Typography__component)>
                  Posted 01 Jan 2019 00:00
                </ForwardRef(Typography__component)>
                <ForwardRef(Typography__styledComponent)>
                  test
                </ForwardRef(Typography__styledComponent)>
                <ForwardRef(Typography__component)>
                  test description
                </ForwardRef(Typography__component)>
                <TagList
                  color="textPrimary"
                  maxTags={3}
                  tags={
                    Array [
                      "test",
                    ]
                  }
                />
              </ForwardRef(ResourceRowWithImage__Content)>,
              "/test/123/a",
            ],
            Array [
              <UserAvatarComponent
                avatar="test avatar"
                cardType="ARTICLE"
                fullWidth={true}
                imageURL="test url"
                userId="0x"
                username="jimi"
              />,
              "/community/0x",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      ownerType="COMMUNITY"
      resourceType="ARTICLE"
      tags={
        Array [
          "test",
        ]
      }
      title="test"
      userAvatar="test avatar"
      userId="0x"
      username="jimi"
    >
      <ResourceRowWithImage__ResourceRow
        key="123"
      >
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "ResourceRowWithImage__ResourceRow-tpmrbu-0",
                "isStatic": false,
                "lastClassName": "hVSttN",
                "rules": Array [
                  "display:flex;width:",
                  [Function],
                  "px;height:195px;background-color:",
                  [Function],
                  ";border-radius:4px;box-shadow:0 0 4px 0 rgba(0,0,0,0.11);",
                ],
              },
              "displayName": "ResourceRowWithImage__ResourceRow",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "ResourceRowWithImage__ResourceRow-tpmrbu-0",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <div
            className="ResourceRowWithImage__ResourceRow-tpmrbu-0 hVSttN"
          >
            <ResourceRowWithImage__Container>
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "ResourceRowWithImage__Container-tpmrbu-2",
                      "isStatic": false,
                      "lastClassName": "gCKJgL",
                      "rules": Array [
                        "display:flex;width:100%;flex-direction:column;padding:",
                        [Function],
                        "px ",
                        [Function],
                        "px;",
                      ],
                    },
                    "displayName": "ResourceRowWithImage__Container",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "ResourceRowWithImage__Container-tpmrbu-2",
                    "target": "section",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <section
                  className="ResourceRowWithImage__Container-tpmrbu-2 gCKJgL"
                >
                  <ResourceRowWithImage__Divider>
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "ResourceRowWithImage__Divider-tpmrbu-3",
                            "isStatic": false,
                            "lastClassName": "cDKOiH",
                            "rules": Array [
                              "width:100%;margin-top:auto;margin-bottom:",
                              [Function],
                              "px;background-color:",
                              [Function],
                              ";height:2px;",
                            ],
                          },
                          "displayName": "ResourceRowWithImage__Divider",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "ResourceRowWithImage__Divider-tpmrbu-3",
                          "target": "div",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                    >
                      <div
                        className="ResourceRowWithImage__Divider-tpmrbu-3 cDKOiH"
                      />
                    </StyledComponent>
                  </ResourceRowWithImage__Divider>
                  <ResourceRowWithImage__Content>
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "ResourceRowWithImage__Content-tpmrbu-1",
                            "isStatic": false,
                            "lastClassName": "hBgJZt",
                            "rules": Array [
                              "display:flex;flex-direction:column;> *:not(:last-child){margin-bottom:",
                              [Function],
                              "px;}> *:nth-child(3){margin-bottom:0px;}",
                            ],
                          },
                          "displayName": "ResourceRowWithImage__Content",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "ResourceRowWithImage__Content-tpmrbu-1",
                          "target": "div",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                    >
                      <div
                        className="ResourceRowWithImage__Content-tpmrbu-1 hBgJZt"
                      />
                    </StyledComponent>
                  </ResourceRowWithImage__Content>
                </section>
              </StyledComponent>
            </ResourceRowWithImage__Container>
          </div>
        </StyledComponent>
      </ResourceRowWithImage__ResourceRow>
    </ResourceRowWithImage>
  </ThemeProvider>
</Provider>
`;
